### PR TITLE
Added Group `resolve`

### PR DESCRIPTION
**Pull request by RichardLitt**

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -1700,6 +1700,130 @@ root will not be listable, as it is virtual. Access known paths directly.
 ## gc
 
 # Group resolve
+Resolve the value of names to IPFS.
+
+## resolve [GET /resolve{?arg}{&resolve}]
+There are a number of mutable name protocols that can link among
+themselves and into IPNS.  For example IPNS references can (currently)
+point at IPFS object, and DNS links can point at other DNS links, IPNS
```
Comment (dignifiedquire) [inline apiary.apib]: at **an** IPFS object

File: apiary.apib [hunk 2]
```diff
@@ -1700,6 +1700,130 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+entries, or IPFS objects. This command accepts any of these
+identifiers and resolves them to the referenced item.
+
++ Parameters
+    + arg (string, required) - The name to resolve.
+    + recursive (boolean, optional) - Resolve until the result is an IPFS name
```
Comment (dignifiedquire) [inline apiary.apib]: Default value?

File: apiary.apib [hunk 3]
```diff
@@ -1700,6 +1700,130 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+    + Body
+
+        ```
+        {
+          "Message": "invalid ipfs ref path",
+          "Code": 0
+        }
+        ```
+
++ Request With Argument
```
Comment (dignifiedquire) [inline apiary.apib]: Would be good to have an example with `recursive=true`

File: apiary.apib (hunk 1 above)
Comment (RichardLitt) [inline apiary.apib]: https://github.com/ipfs/go-ipfs/pull/2363

File: apiary.apib (hunk 2 above)
Comment (RichardLitt) [inline apiary.apib]: No idea. The Wireshark HTTP request doesn't have a flag unless provided. @diasdavid?

File: apiary.apib (hunk 3 above)
Comment (RichardLitt) [inline apiary.apib]: Agreed. Not sure how this works. @diasdavid?

Comment (daviddias) [inline apiary.apib]: The output would be the same :)

File: apiary.apib [hunk 4]
```diff
@@ -1700,6 +1700,130 @@ root will not be listable, as it is virtual. Access known paths directly.
...
+        ```
+        curl -i "http://localhost:5001/api/v0/resolve?arg="
+        ```
+
++ Response 500
+
+    + Headers
+
+        ```
+        Access-Control-Allow-Headers: X-Stream-Output, X-Chunked-Output
```
Comment (daviddias) [inline apiary.apib:2312]: so, here is output and not ouptut? Did you corrected it manually @RichardLitt ?

Comment (RichardLitt) [inline apiary.apib:2312]: I'm sorry, can you clarify what you are asking?

File: apiary.apib (hunk 2 above)
Comment (RichardLitt) [inline apiary.apib]: Looked at code. https://github.com/ipfs/go-ipfs/pull/2410

**Comment by RichardLitt** [comment]
Got a LGTM from @diasdavid.

**Comment by daviddias** [comment]
LGTM :)

**Pull request merged**